File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // initialize hashset with elements from nums
        unordered_set<int> numSet(nums.begin(),nums.end());

        // initialize result variable
        int longest = 0;

        // loop over hashset
        for(int x:numSet){
            // if nums[i]-1 doesnt exist
            if(numSet.find(x-1) == numSet.end()){
                // initialize tempCount = 1
                int tempCount = 1;
                // run loop while nums[i]+1 exist in nums
                while(numSet.find(x+tempCount) != numSet.end()){
                    // increment tempCount
                    tempCount++;
                }
                // set longest to max b/w longest and tempCount
                longest = max(longest, tempCount);
            }
        }
        // return longest
        return longest;
    }
};

File: Data Structures & Algorithms/majority-element/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let result = nums[0], count = 1;
        for(let i=1;i<nums.length;i++){
            if(nums[i]==result) count++;
            else count--;

            if(count<0){
                result = nums[i];
                count = 1
            }
        }
        return result;
    }
}
